File: 179.largest_number/largestNumber.js
/**
 * @param {number[]} nums
 * @return {string}
 */
var largestNumber = function(nums) {
    var num = [];
    for (var i of nums) {
        num.push(i.toString());
    }
    num.sort(function(a,b){
        if (a+b >= b+a){
            return -1;
        }else{
            return 1;
        }
    });
    var res = num.join("");
    //remove the 0s on the left
    for(var j of res){
        if (j != "0") break;
        res = res.substring(1);
    }
    return res || "0";
};

console.log(largestNumber([0,0]));

File: 179.largest_number/largestNumber.py
class Solution:
    # @param {integer[]} nums
    # @return {string}
    def largestNumber(self, nums):
        num = [str(x) for x in nums]
        num.sort(cmp=lambda x, y: cmp(y+x, x+y))
        return ''.join(num).lstrip('0') or '0'

s = Solution()
print s.largestNumber([3, 30, 34, 5, 9])